assets/js/editor.js

```diff
@@ -1,811 +1,613 @@
+                error = "Erreur " + resp.status + " " + resp.statusText + " : " + "'" + resp.responseText.split("\n")[0] + "'";
             } else if(resp.readyState === 0 && resp.statusText === "error") {
                 error = "Oups ! Impossible de se connecter au serveur.";
             }
-
-            printErr(error);
-
-            $editor.val($editor.val().replace(new RegExp(mdWaitingRegexp), ''));
+            onError(error);
         });
-    });
-    return true;
-}
+    };
 
-(function($, undefined){
-    "use strict";
-    var csrf = $("input[name=csrfmiddlewaretoken]").val();
-    var dragEventNumber = 0;
-    $(".md-editor").on("keydown", function(e){
-        // the message is submitted if the user is pressing Ctrl or Cmd with Enter and isn't pressing Alt or Shift
-        if((e.ctrlKey || e.metaKey) && e.which === 13 && !e.altKey && !e.shiftKey){
-            e.preventDefault();
+    $(".md-editor").each(function(){
+        var formEditor = $(this).closest("form");
+
+        var customMarkdownParser = function(plainText) {
+            var result;
+            $.ajax({
+                url: formEditor.attr("action"),
+                type: "POST",
+                data: {
+                    "csrfmiddlewaretoken": csrf,
+                    "text": plainText,
+                    "last_post": "",
+                    "preview": "preview"
+                },
+                success: function(data){
+                    result = data;
+                },
+                async: false
+            });
+            return result;
+        };
 
-            $(".message-submit > button[name=answer]").click();
-        }
-    }).on("dragenter", function(e) {
-        dragEventNumber++;
-        e.preventDefault();
-        if (dragEventNumber === 1) {
-            $(e.target).addClass("selected");
-        }
-    }).on("dragover", function(e) {
-        e.preventDefault();
-    }).on("dragleave", function(e) {
-        dragEventNumber--;
-        e.preventDefault();
-        if (dragEventNumber === 0) {
-            $(e.target).removeClass("selected");
-        }
-    }).on("drop", function (e) {
-        e.preventDefault();
-        uploadImage(e, "dataTransfer", csrf);
-        $(e.target).removeClass("selected");
-        dragEventNumber = 0;
-    }).on("paste", function(e) {
-        var clipboard = e.originalEvent.clipboardData;
-        var data = clipboard.files || clipboard.items;
-        if (!data.length) { //                  ^^^^^ Edge
-            return;
-        }
-        if (uploadImage(e, "clipboardData", csrf)) {
-            e.preventDefault();
-        }
+        var currentLocation = document.location.href.split("/");
+
+        var easyMDE = new EasyMDE({
+                element: this,
+                autosave: {
+                    enabled: true,
+                    uniqueId: this.id+currentLocation[currentLocation.length - 1],
```

Comment:
Je suggère d'essayer ça :

```suggestion
                    uniqueId: window.location.pathname + "@" + this.getAttribute("name"),
```

Je me suis inspiré de [`assets/js/editor-persistence.js`](https://github.com/zestedesavoir/zds-site/blob/dev/assets/js/editor-persistence.js#L51-L52) qui fonctionne bien.

Reply:
Je confirme, ma proposition fonctionne correctement ! Bon par contre EasyMDE crée une entrée `localStorage` par éditeur, même si celui-ci est vide ; notre truc maison était plus optimisé :D

Reply:
Au pire il suffit de clean les clés EasyMDE vide.